docs/Unicode.md

```diff
@@ -0,0 +1,129 @@
+ differently.
+
+```
+import unicodedata
+
+def compare_strs(s1, s2):
+    def NFD(s):
+        return unicodedata.normalize('NFD', s)
+
+    return NFD(s1) == NFD(s2)
+
+single_char = 'ê'
+multiple_chars = '\N{LATIN SMALL LETTER E}\N{COMBINING CIRCUMFLEX ACCENT}'
+print('length of first string=', len(single_char))
+print('length of second string=', len(multiple_chars))
+print(compare_strs(single_char, multiple_chars))
+```
+Returns:
+```
+length of first string= 1
+length of second string= 2
+True
+```
+
+Documentation: https://docs.python.org/3/howto/unicode.html
+
+### Go
+
+Go encodes string as byte array and thus the `length` is determined by the number of bytes 
+instead of the number of characters.
+
+Documentation/Source:
+- https://golang.org/pkg/unicode/
+- https://coderwall.com/p/k7zvyg/dealing-with-unicode-in-go
+
+### Ruby
+
+Tracks `length` by number of bytes. Apparently does not support unicode very well. (Seems like a 
+messed up version of Python's `normalize()` somewhere in the background?)
+
+Documentation/Source: 
+- https://blog.daftcode.pl/fixing-unicode-for-ruby-developers-60d7f6377388
+- https://www.honeybadger.io/blog/ruby-s-unicode-support/
+
+### mySQL
+
+MySQL supports these Unicode character sets:
+- utf8mb4: A UTF-8 encoding of the Unicode character set using one to four bytes per character.
+- utf8mb3: A UTF-8 encoding of the Unicode character set using one to three bytes per character.
+- utf8: An alias for utf8mb3.
+- ucs2: The UCS-2 encoding of the Unicode character set using two bytes per character.
+- utf16: The UTF-16 encoding for the Unicode character set using two or four bytes per character. Like ucs2 but with an extension for supplementary characters.
+- utf16le: The UTF-16LE encoding for the Unicode character set. Like utf16 but little-endian rather than big-endian.
+- utf32: The UTF-32 encoding for the Unicode character set using four bytes per character.
+
+Documentation/Source: 
+- https://dev.mysql.com/doc/refman/8.0/en/charset-unicode.html
+- https://dev.mysql.com/doc/refman/8.0/en/charset-unicode-utf8mb4.html
+
+### Oracle
+
+Does support unicode.
+
+The `NCHAR` datatype stores data encoded as Unicode. The column length specified for the `NCHAR` and 
+`NVARCHAR2` columns is always the number of characters instead of the number of bytes.
+
+"The lengths of the SQL `NCHAR` datatypes are defined as number of characters. This is the same as 
+the way they are treated when using wchar_t strings in Windows C/C++ programs. This reduces programming 
+complexity."
+
+Documentation: https://docs.oracle.com/cd/B19306_01/server.102/b14225/ch6unicode.htm
+
+### MongoDB
+
+Has functions to deal by both bytes and by code points. For those functions that work through bytes, returns
+an error if user requests something that does not split a character evenly according to its number of bytes.
+
+Example: `{ $substrBytes: [ "cafétéria", 7, 3 ] }` \
```

Comment:
hmm...this is interesting.  gives the user some feedback but it's up to the user to know character sizes?  are there functions to help them around this?

Reply:
There is a `stringLenBytes` that user can utilize to find the number of bytes in a string? And they can find the number of bytes in 1 code point by entering a 1 character string. 

Other than that I'm not sure. Maybe they just expected more info to be easy to Google?